### PR TITLE
Player movement

### PR DESCRIPTION
**Pull request by 1095795**
rotation works properly gravity also works, needs a CharacterController to work.

### CODE REVIEW

File: Assets/Scripts/PlayerMovementScript.cs
Comment (ilhs1096069) [inline Assets/Scripts/PlayerMovementScript.cs:1]: Explained this earlier but basically make sure any public variables are instead `[SerializedField]` private variables

File: Assets/Scripts/PlayerMovementScript.cs [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+{
+    // Movement variables
+    private Vector2 _input;
+    private CharacterController _characterController;
+    private Vector3 _direction;
+    public float _moveSpeed;
+
+    // Gravity variables
+    private float _gravity = -9.81f;
+    public float gravityMultiplyer = 3.0f;
```
Comment (ilhs1096069) [inline Assets/Scripts/PlayerMovementScript.cs]: you spelt multiplier wrong :)

Comment (ilhs1096069) [reply Assets/Scripts/PlayerMovementScript.cs]: also why is this public but you have a serialized field down below?

File: Assets/Scripts/PlayerMovementScript.cs [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+    void Start()
+    {
+        if (_characterController == null)
+        {
+            _characterController = GetComponent<CharacterController>();
+        }
+    }
+
+    // Update is called once per frame
+    void Update()
```
Comment (ilhs1096069) [inline Assets/Scripts/PlayerMovementScript.cs:37]: I'd recommend changing this into a FixedUpdate() just to make sure the player isn't moving faster or slower depending on frame rate. Make sure all references to Time.deltaTime are changed to Time.fixedDeltaTime

File: Assets/Scripts/PlayerMovementScript.cs [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+    {
+
+        _input = new Vector2(Input.GetAxis("Horizontal"), Input.GetAxis("Vertical")).normalized;
+        _direction = new Vector3(_input.x, 0f, _input.y);
```
Comment (ilhs1096069) [inline Assets/Scripts/PlayerMovementScript.cs:40]: We should probably in the future make sure the direction is based on the camera so pressing forward moves you away from the camera

**Review by ilhs1096069 — CHANGES_REQUESTED** [review]
Good code just some changes needed here and there

**Comment by ilhs1096069** [comment]
Did some cleanup to the code since Nathan didn't

**Review by roguefirework — APPROVED** [review]
Looks good to me

**Review by ilhs1096069 — APPROVED** [review]

**Pull request merged**